internal/cli/login.go

```diff
@@ -104,51 +106,51 @@ func (o *LoginOptions) Validate(args []string) error {
 		return fmt.Errorf("API URL is not a valid URL")
 	}
 
-	clientConfig, err := o.getClientConfig(args[0])
+	o.clientConfig, err = o.getClientConfig(args[0])
 	if err != nil {
 		return err
 	}
-	o.clientConfig = clientConfig
-	authConfig, err := o.getAuthConfig()
+	o.authConfig, err = o.getAuthConfig()
 	if err != nil {
 		return fmt.Errorf("failed to get auth info: %w", err)
 	}
-	if authConfig == nil || authConfig.AuthType == "" {
+	if o.authConfig == nil {
 		// auth disabled
 		return nil
 	}
-	o.authConfig = authConfig
-
-	if strIsEmpty(o.AccessToken) && strIsEmpty(o.Password) && strIsEmpty(o.Username) && !o.Web {
-		fmt.Println("You must provide one of the following options to log in:")
-		fmt.Println("  --token=<token>")
-		fmt.Println("  --username=<username> and --password=<password>")
-		fmt.Print("  --web (to log in via your browser)\n\n")
-		if o.authConfig.AuthType == "k8s" && !strIsEmpty(o.authConfig.AuthURL) {
-			oauth2 := login.NewK8sOAuth2Config(o.AuthCAFile, o.ClientId, o.authConfig.AuthURL, o.InsecureSkipVerify)
-			oauthConfig, err := oauth2.GetOAuth2Config()
-			if err != nil {
-				return fmt.Errorf("could not get oauth config: %w", err)
-			}
-			fmt.Printf("To obtain the token, you can visit %s/request\n", oauthConfig.TokenEndpoint)
-			fmt.Printf("Then login via \"flightctl login %s --token=<token>\"\n\n", args[0])
-			fmt.Printf("Alternatively, use \"flightctl login %s --web\" to login via your browser\n\n", args[0])
-		}
-		return fmt.Errorf("not enough options specified")
+	o.authProvider, err = o.getAuthProvider()
```

Comment:
Shouldn't creation of data members be external to validation function ?

Reply:
I've split the previous validation func to 3 parts
 - Validate
 - Init (creates the data members)
 - ValidateAuthProviders (this validation does the auth provider specific validation that needs the Init to succeed) 

WDYT ?

Reply:
Seems OK.